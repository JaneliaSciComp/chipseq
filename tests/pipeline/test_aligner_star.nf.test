nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_aligner_star"
    tag "pipeline"

    test("SPT5_T0 / SPT5_T15 / SPT5_INPUT") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            aassertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert new File("$outputDir/bwa/").exists() },
                { assert new File("$outputDir/fastqc/").exists() },
                { assert new File("$outputDir/genome/").exists() },
                { assert new File("$outputDir/igv/").exists() },
                { assert new File("$outputDir/multiqc/").exists() },
                { assert new File("$outputDir/pipeline_info/").exists() },
                { assert new File("$outputDir/trimgalore/").exists() }
            )
        }

    }

}
