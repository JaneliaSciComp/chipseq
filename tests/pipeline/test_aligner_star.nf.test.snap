{
    "genome": {
        "content": [
            "genome.fa:md5,4bad9f4b18056156a81f7f952abbe125",
            "genome.fa.fai:md5,6f4c0ce5258e6948135ad006e1f9ee1b",
            "genome.fa.sizes:md5,e122f8021a5ef14104c32785cb9b7adc",
            "genome.include_regions.bed:md5,58b27f8faf81d116a596c37be421ce01"
        ],
        "timestamp": "2023-06-23T14:22:15+0000"
    },
    "software_versions": {
        "content": [
            "{ANNOTATE_BOOLEAN_PEAKS={sed=4.7}, BAMTOOLS_FILTER={bamtools=2.5.2, samtools=1.15.1}, BAM_REMOVE_ORPHANS={samtools=1.15.1}, BEDTOOLS_GENOMECOV={bedtools=2.30.0}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, CUSTOM_GETCHROMSIZES={getchromsizes=1.16.1}, DEEPTOOLS_COMPUTEMATRIX={deeptools=3.5.1}, DEEPTOOLS_PLOTFINGERPRINT={deeptools=3.5.1}, DEEPTOOLS_PLOTHEATMAP={deeptools=3.5.1}, DEEPTOOLS_PLOTPROFILE={deeptools=3.5.1}, FASTQC={fastqc=0.11.9}, FRIP_SCORE={bedtools=2.30.0, samtools=1.15.1}, GENOME_BLACKLIST_REGIONS={bedtools=2.30.0}, GTF2BED={perl=5.26.2}, HOMER_ANNOTATEPEAKS_CONSENSUS={homer=4.11}, HOMER_ANNOTATEPEAKS_MACS2={homer=4.11}, IGV={python=3.8.3}, MACS2_CALLPEAK={macs2=2.2.7.1}, MACS2_CONSENSUS={python=3.10.0, r-base=4.1.1}, PHANTOMPEAKQUALTOOLS={phantompeakqualtools=1.2.2}, PICARD_COLLECTMULTIPLEMETRICS={picard=3.0.0}, PICARD_MARKDUPLICATES={picard=3.0.0}, PICARD_MERGESAMFILES={picard=3.0.0}, PLOT_HOMER_ANNOTATEPEAKS={r-base=4.0.3}, PLOT_MACS2_QC={r-base=4.0.3}, SAMPLESHEET_CHECK={python=3.8.3}, SAMTOOLS_FLAGSTAT={samtools=1.16.1}, SAMTOOLS_IDXSTATS={samtools=1.16.1}, SAMTOOLS_INDEX={samtools=1.16.1}, SAMTOOLS_SORT={samtools=1.16.1}, SAMTOOLS_STATS={samtools=1.16.1}, STAR_ALIGN={star=2.6.1d}, STAR_GENOMEGENERATE={star=2.6.1d}, SUBREAD_FEATURECOUNTS={subread=2.0.1}, TRIMGALORE={cutadapt=3.4, trimgalore=0.6.7}, UCSC_BEDGRAPHTOBIGWIG={ucsc=377}, Workflow={nf-core/chipseq=2.1.0dev}}"
        ],
        "timestamp": "2023-06-23T14:22:15+0000"
    },
    "bwa/mergedLibrary/": {
        "content": [
            "SPT5_INPUT_REP1.mLb.clN.sorted.bam:md5,8daf8d3b4adf6ad8e93d3bacf4efe059",
            "SPT5_INPUT_REP2.mLb.clN.sorted.bam:md5,db261844fe9e5f20b250f32a99dbf88a",
            "SPT5_T0_REP1.mLb.clN.sorted.bam:md5,737fa3fdd38998777f1c73f4d1c3eaf7",
            "SPT5_T0_REP2.mLb.clN.sorted.bam:md5,fb54e9065fe338091c2880640f9d202a",
            "SPT5_T15_REP1.mLb.clN.sorted.bam:md5,eb89b7f87024ec83066f2fe4bee106ce",
            "SPT5_T15_REP2.mLb.clN.sorted.bam:md5,80fac2e96d44fe0f33a2bdd98b7a08a4"
        ],
        "timestamp": "2023-06-23T14:22:15+0000"
    }
}